spark/src/main/java/gaffer/spark/GraphConfig.java

```diff
@@ -0,0 +1,512 @@
+/*
+ * Copyright 2016 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gaffer.spark;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.accumulo.core.client.AccumuloException;
+import org.apache.accumulo.core.client.AccumuloSecurityException;
+import org.apache.accumulo.core.client.ClientConfiguration;
+import org.apache.accumulo.core.client.Connector;
+import org.apache.accumulo.core.client.Instance;
+import org.apache.accumulo.core.client.IteratorSetting;
+import org.apache.accumulo.core.client.ZooKeeperInstance;
+import org.apache.accumulo.core.client.mapreduce.AccumuloInputFormat;
+import org.apache.accumulo.core.client.mapreduce.lib.impl.InputConfigurator;
+import org.apache.accumulo.core.client.security.tokens.PasswordToken;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.security.Authorizations;
+import org.apache.hadoop.conf.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import gaffer.accumulostore.utils.IteratorSettingBuilder;
+import gaffer.accumulostore.utils.Pair;
+import gaffer.accumulostore.AccumuloStore;
+import gaffer.accumulostore.key.AccumuloKeyPackage;
+import gaffer.accumulostore.key.exception.IteratorSettingException;
+import gaffer.accumulostore.key.exception.RangeFactoryException;
+import gaffer.accumulostore.operation.impl.GetEdgesBetweenSets;
+import gaffer.accumulostore.operation.impl.GetElementsInRanges;
+import gaffer.data.element.Element;
+import gaffer.data.element.function.ElementFilter;
+import gaffer.data.elementdefinition.view.View;
+import gaffer.data.elementdefinition.view.ViewElementDefinition;
+import gaffer.function.simple.filter.IsLessThan;
+import gaffer.function.simple.filter.IsMoreThan;
+import gaffer.operation.data.EdgeSeed;
+import gaffer.operation.data.EntitySeed;
+import gaffer.operation.impl.get.GetRelatedElements;
+import gaffer.store.StoreException;
+
+/**
+ * A custom configuration class that sets the configuration for MapReduce Jobs
+ * on an Accumulo Graph.
+ */
+public class GraphConfig {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GraphConfig.class);
+
+    // Connection to Accumulo
+    protected Connector connector;
+    protected Authorizations authorizations;
+
+    // View on the graph
+    protected boolean rollUpOverTimeAndVisibility;
```

Comment:
In Gaffer2 we refer to this 'rollUpOverTimeAndVisibility' as 'summarise'. This summarisation is now not limited to time and visibility, it will summarise over all properties that have been store in Accumulo's Key.